src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/logging/AppInsights.java

```diff
@@ -0,0 +1,71 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.logging;
+
+import com.microsoft.applicationinsights.TelemetryClient;
+import com.microsoft.applicationinsights.telemetry.Duration;
+import com.microsoft.applicationinsights.telemetry.RequestTelemetry;
+import com.microsoft.applicationinsights.web.internal.RequestTelemetryContext;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import static java.lang.System.currentTimeMillis;
+
+@Aspect
+@Component
+public class AppInsights {
+
+    private static final Logger log = LoggerFactory.getLogger(AppInsights.class);
+
+    private final TelemetryClient telemetryClient;
+
+    public AppInsights(TelemetryClient telemetryClient) {
+        this.telemetryClient = telemetryClient;
+    }
+
+    @Around("@annotation(org.springframework.scheduling.annotation.Scheduled)")
+    public void aroundSchedule(ProceedingJoinPoint joinPoint) throws Throwable {
+        RequestTelemetryContext requestTelemetry =  new RequestTelemetryContext(currentTimeMillis(), null);
+        boolean success = false;
+
+        try {
+            joinPoint.proceed();
+
+            success = true;
+        } finally {
+            handleRequestTelemetry(requestTelemetry, joinPoint.getTarget().getClass().getSimpleName(), success);
+        }
+    }
+
+    private void handleRequestTelemetry(
+        RequestTelemetryContext requestTelemetryContext,
+        String caller,
+        boolean success
+    ) {
+        String requestName = "Schedule /" + caller;
+
+        RequestTelemetry requestTelemetry = requestTelemetryContext.getHttpRequestTelemetry();
+        if (requestTelemetry != null) {
+            log.info(
+                "NO NULL request requestName: {}, start ticks {}, currentTimeMillis: {}, requestID: {}",
+                requestName,
+                requestTelemetryContext.getRequestStartTimeTicks(),
+                currentTimeMillis(),
+                requestTelemetryContext.getHttpRequestTelemetry().getId()
+            );
```

Comment:
do we need extra log? this is purely for request telemetry why we should over-shadow with extra trace?

Reply:
I will remove, just checking the test pod logs.

Reply:
no need logs - those will be requests. that's the point ;)